### ISSUE TITLE
Descriptive email body

### ISSUE THREAD

**Issue by kapkic** [CausalBenchOrg/CausalBench-Causal_Explanation#12]
Our current email implementation sends the report to the user without any text in the body regarding the experiment itself.
In cases where the explanation algorithm does not work due to insufficient data, we should send the insufficiency information in email body as well, this would save the users a click, and we won't have to attach PDF's.

<img width="695" height="336" alt="Image" src="https://github.com/user-attachments/assets/eee8f2b1-17df-4e28-a10e-e3f1e43c4cc3" />

<img width="511" height="173" alt="Image" src="https://github.com/user-attachments/assets/5dcebf85-194e-4854-bef8-8b36f79a0e66" />

**Comment by AbhinavGor** [comment]
@kapkic do you have any proposed email body?

**Comment by AbhinavGor** [comment]
## Proposed Fix (Simple)

  ### Root Cause

  `lambda_function.py` always sends the static string from `CAUSAL_ANALYSIS_EMAIL_BODY`:
  > "Please find the causal analysis results in the pdf document attached to this email."

  No experiment context, no failure details — regardless of outcome.

  Two sub-problems:
  1. **Success case**: Email body has zero context (outcome metric, # experiments, top effects found)
  2. **Failure/insufficient data case**: When analysis yields empty or NaN effects (too few rows, no hyperparameter variation), email is still generic with no explanation

  There is also a hidden crash: if all downloads fail, `raw_df` is undefined → `NameError` at `causal_analysis_helper.py:182` → exception propagates → email is never sent at all.

  ---

  ### Proposed Changes

  #### 1. `causal_analysis_helper.py` — Surface data quality metadata

  - Initialize `raw_df = pd.DataFrame()` before the download block to prevent `NameError` crash when no files download successfully
  - Add the following fields to the returned `yaml_results` dict:
    - `experiment_count` — total rows after cleaning
    - `hyperparameter_count` — number of hyperparameters analyzed
    - `insufficient_data` — boolean flag
    - `insufficient_data_reason` — human-readable string explaining why

    Insufficient data conditions and their reasons:

    | Condition | Reason string |
    |-----------|---------------|
    | No files downloaded | `"No data files could be downloaded from provided URLs"` |
    | `total_experiments < 2` | `"Too few experiments after data cleaning (need ≥ 2)"` |
    | `len(hyperparameters) == 0` | `"No hyperparameters with sufficient variation found"` |
    | All effects are `nan` | `"Causal effects could not be estimated (insufficient variation in data)"` |

  #### 2. `lambda_function.py` — Dynamic email body

  Add a `build_email_body(causal_analysis_results, event)` function that produces a contextual body:

  - **Always includes**: outcome metric, experiment count/summary, filters applied (if any)
  - **On insufficient data**: explains the specific reason and minimum requirements to get results
  - **On success**: lists top 3 causal effects with sign and magnitude

  Replace the static constant on line 52 with a call to `build_email_body()`.

  **Example body (insufficient data):**
  CausalBench+ Causal Analysis Report

  Outcome metric: Time.Duration
  Experiments: Effects on Time.Duration (1 experiments)
  Filters applied: model=resnet50

  INSUFFICIENT DATA: Causal effects could not be computed.
  Reason: Too few experiments after data cleaning (need ≥ 2)

  To get results, run more experiments with varied hyperparameter configurations.
  Minimum requirements: ≥3 data points per variable, ≥2 unique values per hyperparameter.

  Full results are in the attached PDF report.

  **Example body (success):**
  CausalBench+ Causal Analysis Report

  Outcome metric: Time.Duration
  Experiments: Effects on Time.Duration (47 experiments)

  Top causal effects:
    batch_size: +0.8231
    learning_rate: -0.4105
    num_layers: +0.1892

  Full results are in the attached PDF report.

  #### 3. `common_constants.py` — Minor cleanup

  Remove the now-unused `CAUSAL_ANALYSIS_EMAIL_BODY` constant.

  ---

  ### Files Changed

  | File | Change |
  |------|--------|
  | `causal_analysis_helper.py` | Fix `raw_df` crash; add data quality fields to result dict |
  | `lambda_function.py` | Add `build_email_body()`; replace static constant with dynamic call |
  | `common_constants.py` | Remove unused `CAUSAL_ANALYSIS_EMAIL_BODY` |

**Comment by AbhinavGor** [comment]
@kapkic let me know if this sounds good

**Comment by kapkic** [comment]
LGTM
Thanks for the swift action.
